robolectric/src/test/java/org/robolectric/shadows/ShadowInstrumentationTest.java

```diff
@@ -120,4 +123,24 @@ public void concurrentStartService_hasCorrectStartServiceCount() throws Interrup
 
     assertThat(intentCount).isEqualTo(20000);
   }
+
+  @Test
+  public void setInTouchMode_setFalse() {
+    InstrumentationRegistry.getInstrumentation().setInTouchMode(false);
+
+    View decorView =
+        Robolectric.buildActivity(Activity.class).setup().get().getWindow().getDecorView();
+
+    assertThat(decorView.isInTouchMode()).isFalse();
+  }
+
+  @Test
+  public void setInTouchMode_setTrue() {
+    InstrumentationRegistry.getInstrumentation().setInTouchMode(true);
+
+    View decorView =
+        Robolectric.buildActivity(Activity.class).setup().get().getWindow().getDecorView();
```

Comment:
@hoisie what about recommending to use `ActivityScenario` for Robolectric tests to setup `Activity` except tests for `Robolectric.buildActivity`?

Reply:
@utzcoz the main concern that I have w/ ActivityScenario is that I believe it requires all activities to be declared in a manifest. So at that point you have to subclass Activity and then place it in a manifest somewhere. You would also have to hook up aapt to generate the resources APK, which is non-trivial because it would require the 'robolectric' gradle project to us AGP, or we would need to invoke aapt manually. To me this adds quite a bit of overhead if you want to put up a quick unit test for something. Do you have any thoughts on this?

Alternatively we could look into updating ActivityScenario to drop this requirement for Robolectric.

Reply:
Oh, the requirement for `Activity` defined in `AndroidManifest.xml` is a pain to use `ActivityScenario`.  I have noticed some problems about AGP merging problem for test `Activity` defined in test directory at Robolectric and android-est. Sorry, I forgot this requirement. Looks like the place to use `ActivityScenario` in Robolectric firstly is integration_tests.

 If we can drop this requirement of ActivityScenario for Robolectric, it maybe help to push the adoption of `ActivityScenario`(maybe the AGP merging problem for Activity defined in test directory is greater hindrance for adoption).  In Robolectric, we can skip this restriction with shadow API(enable Activity and export Activity if needed) when Activity class defined in final packaged apk.

Reply:
What is the benefit of using ActivityScenario APIs over equivalent Robolectric APIs if it's done in a way that's incompatible with Instrumentation (i.e. it only works in Robolectric anyway)?

Reply:
@paulsowden for tests in the `integration_tests` directory, they are already hooked up to AGP (which includes manifest merging), so it's a lot easier to use `ActivityScenario`. Also I believe that `integration_tests` tend to use fewer Roboelctric-specific APIs, especially the AXT/Espresso tests.

For tests in `shadows` and `robolectric` directories, they are basically Java gradle modules, not hooked up to AGP/manifest merging, so using ActivityScenario would be quite challenging, unless we invoke aapt2 manually.

Reply:
I will file a new issue to track the work for investigating in dropping the requirement of test Activity defined in manifest of `ActivityScenario` for Robolectric today.

Reply:
I guess I don't understand what is the reason you want to drop the requirement when that is already exactly what `ActivityController`/`Robolectric.buildActivity` already does?

Reply:
The main concern reason is to make test code more compatibility between Robolectric and androidx.test. Robolectric has did a lot to run test code written with Espresso and androidx.test APIs, we can do more for it, and let user write once and run on every Android device. There are two related articles about it: http://robolectric.org/blog/2021/10/06/sharedTest/ and https://medium.com/androiddevelopers/write-once-run-everywhere-tests-on-android-88adb2ba20c5. So I recommend Robolectric to use similar methods to write compatibility tests if possible. And this manifest requirement is the biggest difficulty at present prevents further adoption of `ActivityScenario` in Robolectric's itself tests.  IMO, it needs some work to investigate the possibility to drop it.

Reply:
Sure, it makes total sense to prefer compatible APIs, but there's two things that seem backwards:

1. This is a test for a Robolectric shadow, by definition this class only exists in Robolectric, compatibility with Android is no benefit here.
2. It seems backwards to make ActivityScenario behave differently in Robolectric when the goal is to write code that is compatible. Wouldn't it be better to solve this problem by improving the tooling to make it easier to write ActivityScenario tests with proper android manifests than to modify Robolectric's ActivityScenario support in a way that's incompatible with Instrumentation/Android?

Reply:
Good point, I suppose that it would not be reasonable to drop the manifest requirement for Robolectric, as that would prevent ActivityScenario from working across execution environments. In theory if Android would let you instantiate Activities not declared in manifests you could drop that requirement, but I don't see that happening any time soon.

Reply:
Agreed, thanks @paulsowden. Maybe we could ping AGP team to help to fix merging manifest problem for local test by following above suggestions, and users can define test Activity in manifest under test directory. For Robolectric, we only recommend to use `ActivityScenario` for `integration_tests` and related compatibility tests only, not for Robolectric shadow tests.